envpool/sokoban/sokoban_envpool.cc

```diff
@@ -31,7 +33,12 @@ void SokobanEnv::ResetWithoutWrite() {
   current_max_episode_steps_ =
       SafeUniformInt(min_episode_steps, max_episode_steps, gen_);
 
-  world_ = *(level_loader_.GetLevel(gen_));
+  std::pair<std::vector<std::pair<int, SokobanLevel>>::iterator, int> level =
+      level_loader_.GetLevel(gen_);
+  world_ = (*level.first).second;
```

Comment:
`level.first->second`

This would get easier if TaggedSokobanLevel was a struct. Then you can assign .data to `world_` and all works.